src/renderers/testing/ReactTestRenderer.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactTestRenderer
+ */
+
+'use strict';
+
+var ReactDefaultBatchingStrategy = require('ReactDefaultBatchingStrategy');
+var ReactEmptyComponent = require('ReactEmptyComponent');
+var ReactMultiChild = require('ReactMultiChild');
+var ReactHostComponent = require('ReactHostComponent');
+var ReactTestMount = require('ReactTestMount');
+var ReactTestReconcileTransaction = require('ReactTestReconcileTransaction');
+var ReactUpdates = require('ReactUpdates');
+
+var renderSubtreeIntoContainer = require('renderSubtreeIntoContainer');
+
+/**
+ * Drill down (through composites and empty components) until we get a native or
+ * native text component.
+ *
+ * This is pretty polymorphic but unavoidable with the current structure we have
+ * for `_renderedChildren`.
+ */
+function getRenderedHostOrTextFromComponent(component) {
+  var rendered;
+  while ((rendered = component._renderedComponent)) {
+    component = rendered;
+  }
+  return component;
```

Comment:
could you help me understand why this is returning the component when it’s looking at the `_renderedComponent` all the way down? Is it because when there is no more `_renderedComponent` we have reached the bottom of that component hierarchy?